src/Components/IndexScene/IndexScene.tsx

```diff
@@ -118,18 +120,19 @@ export class IndexScene extends SceneObjectBase<IndexSceneState> {
   }
 
   /**
-   * @todo why do we need to manually sync fields, but nothing else?
-   * @param fieldsVariable
+   * @todo why do we need to manually sync fields and levels, but not other ad hoc variables?
```

Comment:
That is a very good question. Shouldn't we use something like `SceneObjectUrlSyncConfig` for this?

Reply:
You'd think, but it doesn't work, the variable is still cached in the scene despite not being in the url params

Reply:
Well actually let me double check, I forgot you can't just add keys, but implementation needs to be added in `getUrlState`, and `updateFromUrl`

Reply:
Yeah, the url sync seems like it would only help if we were storing some state of the IndexScene in the url, but we're having trouble with the state of the $variable filters. I tried adding levels, fields, and even $variables to the dep array and was never able to get anything useful out of the url functions. 🤷